File: SoftUni - Java OOP/Encapsulation/Encapsulation Exercises/_02_Animal_Farm/Chicken.java
package _02_Animal_Farm;

public class Chicken {
    private String name;
    private int age;

    public Chicken(String name, int age) {
        setName(name);
        setAge(age);
    }
    public String toString() {
        return String.format("Chicken %s (age %d) can produce %.2f eggs per day.", this.name, this.age, this.productPerDay());
    }
    private void setName(String name) {
        if((name == null || name.isEmpty() || name.isBlank() || Character.isWhitespace(name.charAt(0))) ) {
            throw new IllegalArgumentException("Name cannot be empty.");
        }
        this.name = name;
    }
    private void setAge(int age) {
        if(age < 0 || age > 15) {
            throw new IllegalArgumentException("Age should be between 0 and 15.");
        }
        this.age = age;
    }
    public double productPerDay() {
        return calculateProductPerDay();
    }

    private double calculateProductPerDay() {
        if(this.age < 6) {
            return 2;
        } else if (this.age < 12) {
            return 1;
        } else {
            return 0.75;
        }
    }
}

File: SoftUni - C# OOP/Encapsulation/Encapsulation - Exercise/02. Animal Farm/02. Animal Farm/Models/Chicken.cs
using System;

namespace AnimalFarm.Models
{
    public class Chicken
    {
        private const int MinAge = 0;
        private const int MaxAge = 15;

        private protected string name;
        private protected int age;

        public Chicken(string name, int age)
        {
            this.Name = name;
            this.Age = age;
        }

        public string Name
        {
            get
            {
                return this.name;
            }

            private protected set
            {
                if (!string.IsNullOrEmpty(value) && !string.IsNullOrWhiteSpace(value))
                {
                    this.name = value;
                }
                else
                {
                    throw new ArgumentException("Name cannot be empty.");
                }
            }
        }

        public int Age
        {
            get
            {
                return this.age;
            }

            private set
            {
                if (value > MinAge && value < MaxAge)
                {
                    this.age = value;
                }
                else
                {
                    throw new ArgumentException($"Age should be between {MinAge} and {MaxAge}.");
                }
            }
        }

        public double ProductPerDay
        {
            get
            {
                return this.CalculateProductPerDay();
            }
        }

        private double CalculateProductPerDay()
        {
            switch (this.Age)
            {
                case 0:
                case 1:
                case 2:
                case 3:
                    return 1.5;
                case 4:
                case 5:
                case 6:
                case 7:
                    return 2;
                case 8:
                case 9:
                case 10:
                case 11:
                    return 1;
                default:
                    return 0.75;
            }
        }
    }
}